zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -788,11 +788,13 @@ pub(crate) struct MulAddWordsGadget<F> {
 }
 
 impl<F: Field> MulAddWordsGadget<F> {
-    pub(crate) fn construct(cb: &mut ConstraintBuilder<F>) -> Self {
-        let a = cb.query_word();
-        let b = cb.query_word();
-        let c = cb.query_word();
-        let d = cb.query_word();
+    pub(crate) fn construct(
+        cb: &mut ConstraintBuilder<F>,
+        a: util::Word<F>,
```

Comment:
I think it would be more consistent if `construct` and `assign` take the cell/values in the same shape.
For example, either both be `a: T, b: T, c: T, d: T`, or `words: [T; 4]`, but not one of each.

Reply:
Yep, I've gone for the second approach everywhere.